### PR TITLE
Release v1.1.0

### PR DESCRIPTION
**Pull request by ghost**
changes:

- added pylint config file
    pylint is a syle checker

- changed: BaseComm is now an abstract class.

- improved documentation

- added pytest for the frame generator

### CODE REVIEW

File: client/comm.py [hunk 1]
```diff
@@ -108,39 +155,17 @@ def _push_frame(self, frame: Frame):
         )
 
     def listen_for(self, comm_listen_for: list) -> None:
-        """
```
Comment (sebastiaan1997) [inline client/comm.py]: Why is this removed?

File: client/comm.py [hunk 2]
```diff
@@ -108,39 +155,17 @@ def _push_frame(self, frame: Frame):
...
-        :param comm_listen_for:
-        :return:
-        """
         self.comm_listen_for = comm_listen_for
 
         if FrameType.ALL in comm_listen_for:
             self.accepts_all = True
 
     def accepts_frame(self, type: FrameType) -> bool:
-        """
```
Comment (sebastiaan1997) [inline client/comm.py]: Why is this removed

File: client/comm.py [hunk 3]
```diff
@@ -108,39 +155,17 @@ def _push_frame(self, frame: Frame):
...
-
-        :param type:
-        :return:
-        """
         if self.accepts_all:
             return True
         return type in self.comm_listen_for
 
     def request(self, type, prio: Priority = Priority.NORMAL) -> None:
-        """
```
Comment (sebastiaan1997) [inline client/comm.py]: Why is this removed?

File: client/comm.py [hunk 4]
```diff
@@ -149,38 +174,15 @@ def request(self, type, prio: Priority = Priority.NORMAL) -> None:
         self._push_frame(frame)
 
     def send(self, frame, prio: Priority = Priority.NORMAL) -> None:
-        """
-        Put a frame on the bus.
```
Comment (sebastiaan1997) [inline client/comm.py]: Why is this removed?

File: client/comm.py (hunk 1 above)
Comment (steviestickman) [reply client/comm.py]: docstrings are moved to base class, removed because if the documentation is kept in two places, one place will become out of date

File: client/comm.py (hunk 2 above)
Comment (steviestickman) [reply client/comm.py]: https://github.com/R2D2-2019/r2d2-python-build/pull/36#discussion_r283974553

File: client/comm.py (hunk 3 above)
Comment (steviestickman) [reply client/comm.py]: https://github.com/R2D2-2019/r2d2-python-build/pull/36#discussion_r283974553

File: client/comm.py (hunk 4 above)
Comment (steviestickman) [reply client/comm.py]: https://github.com/R2D2-2019/r2d2-python-build/pull/36#discussion_r283974553

**Review by sebastiaan1997 — CHANGES_REQUESTED** [review]
Some binaries are included, could these be removed (*.pyc files)?
Some pydoc comments are removed, I don't know why

**Comment by steviestickman** [comment]
the only `*.pyc` files in this diff are those that are deleted, they were in previous versions, but are deleted now.

the docstring comments are moved from the derived clas to the base class.

**Review by sebastiaan1997 — APPROVED** [review]
